### PR TITLE
Migrate 'split' to CallOrComplete

### PR DESCRIPTION
**Pull request by t-wissmann**
The 'split' command was using ArgParse already, but in order to have the
completion generated automatically, I've migrated it to CallOrComplete,
and made SplitMode an enum.

### CODE REVIEW

File: src/frametree.cpp [hunk 1]
```diff
@@ -909,51 +911,74 @@ class SplitMode {
...
-            availableModes.begin(), availableModes.end(),
-            [=](const SplitMode &x){ return x.name[0] == splitType[0]; });
-    if (mode == availableModes.end()) {
-        output << input.command() << ": Invalid alignment \"" << splitType << "\"\n";
-        return HERBST_INVALID_ARGUMENT;
-    }
-    SplitMode m = *mode;
+    auto allSplitModes = SplitMode::modes(align_auto);
+    auto it = allSplitModes.find(mode);
+    // we now that 'mode' must be in the map allSplitModes, but let us stay safe
```
Comment (blarz) [inline src/frametree.cpp:979]: know?

**Review by blarz — COMMENTED** [review]
Just a tiny comment typo (but we need to resolve the merge conflict in commands.cpp anyway :))

File: src/frametree.cpp (hunk 1 above)
Comment (t-wissmann) [reply src/frametree.cpp]: Yes :) I've fixed it, thanks!

Comment (blarz) [reply src/frametree.cpp]: Thanks!

**Review by blarz — APPROVED** [review]

**Comment by blarz** [comment]
This conflicts again :D command.cpp is in high demand these days ;)

**Review by blarz — APPROVED** [review]

File: src/command.cpp [hunk 2]
```diff
@@ -76,8 +71,6 @@ struct {
     const char** list;
 } g_completions[] = {
```
Comment (blarz) [inline src/command.cpp:72]: This is now effectively empty

**Review by blarz — CHANGES_REQUESTED** [review]

File: src/command.cpp (hunk 2 above)
Comment (t-wissmann) [reply src/command.cpp]: I've brought the content back, such that I can remove the unused code from `command.cpp` in a separate commit.

**Review by blarz — APPROVED** [review]

**Pull request merged**